hack/golangci-strict.yaml

```diff
@@ -0,0 +1,38 @@
+# This file configures checks that all new code for Kubernetes is meant to
+# pass, in contrast to .golangci.yaml which defines checks that also the
+# existing code passes.
+
+run:
+  timeout: 30m
+  skip-files:
+    - "^zz_generated.*"
+
+issues:
+  max-same-issues: 0
+  # Excluding configuration per-path, per-linter, per-text and per-source
+  exclude-rules:
+    # exclude ineffassing linter for generated files for conversion
+    - path: conversion\.go
+      linters:
+        - ineffassign
+
+linters:
+  enable: # please keep this alphabetized
```

Comment:
maybe add a comment about how this should only contain linters not already in the standard yaml?

Reply:
Actually, it absolutely should contain those :sweat_smile: 

That way, developers can be sure that `verify-golangci-lint.sh -a -s` catches all linter issues in their code and won't be surprised later when the "normal" config gets applied to it.

Reply:
And when the time comes to integrate it into `pull-kubernetes-verify`, it'll be enough to run verify-golangci-lint.sh only once, with the strict config.

I had a comment about that in the file at some point, but must have lost it when syncing anew with a modified golangci.yaml - I'll bring that comment back.